kinder/ci/tools/update-workflows/templates/workflows/unversioned-kubelet-cm-tasks.yaml

```diff
@@ -0,0 +1,188 @@
+    Verify that the ConfigMap and RBAC objects with the new name exist (UnversionedKubeletConfigMap=true)
+  cmd: /bin/bash
+  args:
+    - -c
+    - |
+      docker exec {{ .vars.clusterName }}-control-plane-1 kubectl --kubeconfig /etc/kubernetes/admin.conf get cm kubelet-config -n kube-system || exit 1
+      docker exec {{ .vars.clusterName }}-control-plane-1 kubectl --kubeconfig /etc/kubernetes/admin.conf get role kubeadm:kubelet-config -n kube-system || exit 1
+      docker exec {{ .vars.clusterName }}-control-plane-1 kubectl --kubeconfig /etc/kubernetes/admin.conf get rolebinding kubeadm:kubelet-config -n kube-system || exit 1
+  timeout: 5m
+- name: join
+  description: |
+    Join the other nodes to the Kubernetes cluster
+  cmd: kinder
+  args:
+  - do
+  - kubeadm-join
+  - --name={{ .vars.clusterName }}
+  - --loglevel=debug
+  - --kubeadm-verbosity={{ .vars.kubeadmVerbosity }}
+  - --copy-certs=auto
+  timeout: 10m
+- name: undo-feature-gate-changes
+  description: |
+    Restore the old ConfigMap and RBAC names, pre-upgrade.
+    With the feature gate set, upgrade should create the objects with the new naming format
+  cmd: /bin/bash
+  args:
+    - -c
+    - |
+      # get the major.minor version
+      VERSION=$(echo {{ .vars.kubernetesVersion }} | cut -d 'v' -f 2 | cut -d '.' -f1,2)
+      echo VERSION=$VERSION
+
+      # duplicate the new naming format objects back to the old naming format objects
+      docker exec {{ .vars.clusterName }}-control-plane-1 bash -c "kubectl --kubeconfig /etc/kubernetes/admin.conf get cm kubelet-config -n kube-system -o=yaml | sed s/'  name: kubelet-config/  name: kubelet-config-$VERSION'/ | kubectl --kubeconfig /etc/kubernetes/admin.conf create -f - " || exit 1
+      docker exec {{ .vars.clusterName }}-control-plane-1 bash -c "kubectl --kubeconfig /etc/kubernetes/admin.conf get role kubeadm:kubelet-config -n kube-system -o=yaml | sed s/'  name: kubeadm:kubelet-config/  name: kubeadm:kubelet-config-$VERSION'/ | sed s/'  - kubelet-config/  - kubelet-config-$VERSION'/ | kubectl --kubeconfig /etc/kubernetes/admin.conf create -f - " || exit 1
+      docker exec {{ .vars.clusterName }}-control-plane-1 bash -c "kubectl --kubeconfig /etc/kubernetes/admin.conf get rolebinding kubeadm:kubelet-config -n kube-system -o=yaml | sed s/'  name: kubeadm:kubelet-config/  name: kubeadm:kubelet-config-$VERSION'/ | kubectl --kubeconfig /etc/kubernetes/admin.conf create -f - " || exit 1
+
+      # delete the new format objects
+      docker exec {{ .vars.clusterName }}-control-plane-1 kubectl --kubeconfig /etc/kubernetes/admin.conf delete cm kubelet-config -n kube-system || exit 1
+      docker exec {{ .vars.clusterName }}-control-plane-1 kubectl --kubeconfig /etc/kubernetes/admin.conf delete role kubeadm:kubelet-config -n kube-system || exit 1
+      docker exec {{ .vars.clusterName }}-control-plane-1 kubectl --kubeconfig /etc/kubernetes/admin.conf delete rolebinding kubeadm:kubelet-config -n kube-system || exit 1
```

Comment:
not particularly happy with these long exec lines, but they work, as they are...
did not want to add specific code in kinder and then remove it once the new FG is GA.

Reply:
let's start with this, then we can iterate